mesa/experimental/observer.py

```diff
@@ -0,0 +1,77 @@
+from collections import defaultdict
+
+import pandas as pd
+
+
+class DataCollector:
+    """
+    Example: a model consisting of a hybrid of Boltzmann wealth model and
+    Epstein civil violence.
+    ```
+    def get_citizen():
```

Comment:
There is a problem with this line: since this function is defined within a model's `__init__`, then there is no way to refer *using* the reference of this function later on, when doing further analysis. The only way that makes sense is to define a group dict
`{"citizen":  lambda: model.get_agents_of_type(Citizen)}` that `DataCollector.collect` can use to resolve the named group.